src/test.rs

```diff
@@ -62,7 +61,7 @@ fn allocate_and_free_double_usize() {
     let mut heap = new_heap();
 
     let layout = Layout::from_size_align(size_of::<usize>() * 2, align_of::<usize>()).unwrap();
-    let x = heap.allocate_first_fit(layout.clone()).unwrap();
```

Comment:
We shouldn't just remove the `unwrap`, because it also checked that the allocation succeeded. When we don't check it we risk deallocating a null pointer. Also applies to the other cases where unwraps are removed.

Reply:
Looks like I'll have to add asserts here, or write a function that panics when pointer is null